File: ios/ReactNativeBridge.m
//
//  ReactNativeBridge.m
//  BLEApp
//
//  Created by Vedant Alimchandani on 15/03/24.
//

#import <Foundation/Foundation.h>
#import <React/RCTBridgeModule.h>
#import <React/RCTEventEmitter.h>

@interface RCT_EXTERN_MODULE(ReactNativeBridge, NSObject)
RCT_EXTERN_METHOD(sendStringToReactNative:(NSString *)message)
@end

File: ios/ReactNativeBridge.swift
//
//  ReactNativeBridge.swift
//  BLEApp
//
//  Created by Vedant Alimchandani on 15/03/24.
//

import React

@objc(ReactNativeBridge)
class ReactNativeBridge: NSObject {
    
    @objc static func requiresMainQueueSetup() -> Bool {
        return true
    }
    
    // Method to send a string to React Native
    @objc func sendStringToReactNative(_ message: String) {
        if let bridge = RCTBridge.current() {
            let reactData = ["message": message]
            bridge.enqueueJSCall("RCTDeviceEventEmitter.emit", args: ["TestEvent", reactData])
        }
    }
    
    
}

